### DISCUSSION TITLE
Bug: Angular 14+ using the inject with defaultMock is not working

### DISCUSSION THREAD

**Discussion by KevinBonnoron** [help-me-mom/ng-mocks#5073] [Q&A] [answered]
### Description of the bug

When using MockBuilder, i try to inject a custom InjectionToken with the new "inject" function from angular 14, and trying to mock it globally with a "ngMocks.defaultMock" the test fail with a "No provider for InjectionToken" error. It's working when using the "old way" of doing injection with the constructor and the @Inject decorator or if i provide a mock with the MockBuilder.mock function.

### An example of the bug

Link: https://github.com/KevinBonnoron/ng-mocks-inject-bug

### Expected vs actual behavior

Expected behaviour: The test should pass

Actual behaviour: The test is failing

The global mock for the token is done in src/setupJest.ts

I commented some code and added comments to indicate what parts are working and what aren't.

**Comment by satanTime** [comment, marked as answer]
Hi there, 

you are very close.

Because `ENVIRONMENT_TOKEN` is a dependency, it should be provided in `MockBuilder`:

```ts
beforeAll(() => MockBuilder(AppComponent, [ENVIRONMENT_TOKEN]));
```

Otherwise `ng-mocks` cannot detect it, because of `inject` it's a runtime dependency, whereas before with `@Inject` it was a declaration dependency, which could be detected without executing code.

The correct version of your test: https://codesandbox.io/s/silly-wescoff-xkg4ri?file=/src/test.spec.ts:641-658

**Reply by c-harding** [reply]
Hi @satanTime, how do I do this in all components? Is there a global function? The following combination doesn’t work:

```ts
ngMocks.globalMock(BASE_PATH);
ngMocks.defaultMock(BASE_PATH, () => '/api/test-path');
```

**Comment by crainsaw** [comment]
I stumbled over that as well. Would it make sense to add it to the documentation at https://ng-mocks.sudo.eu/api/ngMocks/defaultMock/ ?
